firebase-common/src/main/java/com/google/firebase/StartupTime.java

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase;
+
+import android.os.SystemClock;
+import androidx.annotation.NonNull;
+import com.google.auto.value.AutoValue;
+import javax.annotation.Nullable;
+
+/**
+ * Represents the time at which Firebase began initialization, both in unix time/epoch milliseconds
+ * and in milliseconds since boot. The absence of a StartupTime indicates an unreliable or
+ * misleading time, such as a launch in direct boot mode. Because of this, StartupTime cannot be
+ * guaranteed to be present, and instead should be optionally depended on, and its absence handled.
+ */
```

Comment:
Nit: please write a concise summary fragment: go/java-practices/javadoc#summary_fragment

Reply:
and please add `@hide` as the last line of the javadoc, this will suppress public javadocs for this type.